### PR TITLE
#157694756 Job running too often

### PR DESCRIPTION
**Pull request by sam-karis**
#### What does this PR do?
- Add an icon on the dashboard to show a job that is run too often
- Sends an email to the user to notify that a job is running too often

#### Description of Task to be completed?
When a job runs too often a user can see an icon on the dashboard and also receive an email notification each time a job run earlier than expected(too often).
#### How should this be manually tested?
- Run the application and click the check link to create a new check
- Run the job too often to receive a notification email and view an icon on the dashboard
#### What are the relevant pivotal tracker stories?
[Job running too often](https://www.pivotaltracker.com/story/show/157694756)

Screenshots
A sample of an icon on the dashboard

image

Sample of the email a user receive

image

### CODE REVIEW

**Comment by kzyangiro** [comment]
Good job @sam-karis . Wonderful logic and output. I would however wish that you change the nagging icon to a different color since the current green color can be mistaken as of an "up" check which is blue in color.

**Comment by sam-karis** [comment]
Hi, @kzyGit thanks for the feedback will take into consideration as I make improvements on the project.

File: hc/api/models.py [hunk 1]
```diff
@@ -85,12 +86,26 @@ def get_status(self):
...
             return "up"
-
         return "down"
 
+    def is_running_too_often(self):
+        now = timezone.now()
+        if not self.last_ping:
+            return False
+
+        now = timezone.now()
```
Comment (Michyjones) [inline hc/api/models.py:105]: Great thought here. I like everything you have done in logic and execution. It looks good to me. The only thing you could work on is to remove the unused line of code. This (now = timezone.now()) is a repetition just one is enough in this function

Comment (sam-karis) [reply hc/api/models.py:105]: @Michyjones  Thanks for the feedback and noted working on removing the repetition.

**Pull request merged**